src/evm/decoder.rs

```diff
@@ -61,12 +63,20 @@ impl TychoStreamDecoder {
         }
     }
 
+    /// Sets the currently known tokens which will be considered during decoding.
+    ///
+    /// Protocol components containing tokens which are not included in this initial list, or
+    /// added when applying deltas, will not be decoded.
     pub async fn set_tokens(&self, tokens: HashMap<Bytes, Token>) {
         let mut guard = self.state.write().await;
         guard.tokens = tokens;
     }
 
-    // Method to register a decoder
+    /// Registers a decoder for a given exchange.
+    ///
+    /// Each of the native protocol states, along with the EVMPoolState, have their own decoders.
+    /// For example, to register a decoder for the `uniswap_v2` exchange, you must call
+    /// this function with `register_decoder::<UniswapV2State>("uniswap_v2")`.
```

Comment:
Would rather I would explain here that this associates a type with an exchanges id. It is required that the type implement `TryFromWithBlock`.